client/my-sites/upgrades/checkout-thank-you/plans-page-your-plan.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import page from 'page';
+
+/**
+ * Internal Dependencies
+ */
+import { renderWithReduxStore } from 'lib/react-helpers';
+import Main from 'components/main';
+import Card from 'components/card';
+import HappinessSupport from 'components/happiness-support';
+import PremiumPlanDetails from './premium-plan-details';
+import BusinessPlanDetails from './business-plan-details';
+import CheckoutThankYouFeaturesHeader from './features-header';
+import CheckoutThankYouHeader from './header';
+import sitesList from 'lib/sites-list';
+import { getPlansBySite } from 'state/sites/plans/selectors';
+import { fetchSitePlans } from 'state/sites/plans/actions';
+import { shouldFetchSitePlans } from 'lib/plans';
+import {
+	isBusiness,
+	isPremium,
+	isFreePlan
+} from 'lib/products-values';
+
+export default function( context ) {
+	const sites = sitesList();
+	const selectedSite = sites.getSelectedSite();
+
+	if ( shouldFetchSitePlans( getPlansBySite( context.store.getState(), selectedSite ), selectedSite.ID ) ) {
+		context.store.dispatch( fetchSitePlans( selectedSite.ID ) );
+	}
+
+	const PlanDetailsComponent = React.createClass( {
```

Comment:
Also, let's just separate controllers from the components to be rendered. The general structure is:

```
- index.js (declares the routes as is the entry point for the whole section)
- controller.js (handles functions to be executed as routes are matched
- main.jsx (is the primary render tree for your area and includes the <Main> component)
```